DeployJenkinsfile

```diff
@@ -20,156 +21,252 @@ pipeline {
+                            echo "Publishing UAA ${APP_VERSION} Artifact to Artifactory"
+                            def uploadSpec = """{
+                                "files": [
+                                    {
+                                        "pattern": "deploy-workspace/cloudfoundry-identity-uaa-${APP_VERSION}.war",
+                                        "target": "MAAXA/org/cloudfoundry/identity/cloudfoundry-identity-uaa/${APP_VERSION}/"
+                                    }
+                                ]
+                            }"""
+                            def buildInfo = buildGeArtServer.upload(uploadSpec)
+                            buildGeArtServer.publishBuildInfo(buildInfo)
+                        }
+                    }
                 }
             }
         }
-        stage('Upload UAA Artifact upon successful deploy of staging artifact') {
-            agent {
-                label 'dind'
+        stage('EKS Deploy')
+        {
+            when 
+            {
+                expression { params.DEPLOY_EKS == true }
             }
-            when {
-                expression { params.DEPLOYMENT_TYPE == 'cf3-staging' }
-            }
-            steps{
-                dir('deploy-workspace') {
-                    unstash 'uaa-war'
-                }
-
-                dir('uaa') {
-                    checkout scm
+            agent{
+                docker {
+                    image "${NODE['IMAGE']}"
+                    label "${NODE['LABEL']}"
+                    args "${NODE['ARGS']}"
                 }
-
-                dir('uaa/iam-k8s-utils') {
-                    git changelog: false, credentialsId: 'github.build.ge.com', poll: false,
-                        url: 'https://github.build.ge.com/predix/iam-k8s-utils.git'
+            }
+            stages
+            {
+                stage('Get GitOps Code Repository')
+                {
+                    steps
+                    {
+                        dir('uaa-k8s-deploy')
+                        {
+                            // Check out repo with Dockerfiles and build/publish script
+                            git changelog: false,
+                                    credentialsId: 'DEVTOOLS_GITHUB_CREDENTIALS',
+                                    poll: false,
+                                    url: 'https://github.devtools.predix.io/predix-security/uaa-k8s-deploy.git',
+                                    branch: 'master'
+                        }
+                    }
                 }
+                stage('Update GitOps Repository')
+                {
+                    when {
+                        not { changeRequest() }
+                    }
+                    steps
+                    {
+                        script
+                        {
+                            if (params.DEPLOYMENT_TYPE == "cf3-release-candidate")
```

Comment:
As agreed, add the list of the EKS environment to DEPLOYMENT_TYPE choice parameter.